### PR TITLE
Savings validation e2e tests

### PR DESCRIPTION
**Pull request by oskarvu**
Closes FRO-723

### CODE REVIEW

File: packages/app/src/features/dialogs/savings/deposit/e2e/mainnet/DepositDAI.test-e2e.ts [hunk 1]
```diff
@@ -70,3 +71,64 @@ test.describe('Deposit DAI on Mainnet', () => {
...
+
+    savingsPage = new SavingsPageObject(page)
+    await savingsPage.clickDepositButtonAction('DAI')
+
+    depositDialog = new SavingsDialogPageObject({ page, type: 'deposit' })
+  })
+
+  test('displays validation error', async () => {
+    await depositDialog.fillAmountAction(200)
+    await depositDialog.expectAssetInputError(depositValidationIssueToMessage['exceeds-balance'])
```
Comment (yivlad) [inline packages/app/src/features/dialogs/savings/deposit/e2e/mainnet/DepositDAI.test-e2e.ts:100]: I think we should hardcode the error messages here. It's ok if we'll get "notified" when the text changes

File: packages/app/src/features/dialogs/savings/deposit/e2e/mainnet/DepositDAI.test-e2e.ts [hunk 2]
```diff
@@ -70,3 +71,64 @@ test.describe('Deposit DAI on Mainnet', () => {
...
+    await depositDialog.fillAmountAction(0)
+    await depositDialog.expectAssetInputError(depositValidationIssueToMessage['value-not-positive'])
```
Comment (yivlad) [inline packages/app/src/features/dialogs/savings/deposit/e2e/mainnet/DepositDAI.test-e2e.ts:102]: Same as above

File: packages/app/src/features/dialogs/savings/deposit/e2e/mainnet/DepositDAI.test-e2e.ts (hunk 1 above)
Comment (oskarvu) [reply packages/app/src/features/dialogs/savings/deposit/e2e/mainnet/DepositDAI.test-e2e.ts:100]: Was following convention @krzkaczor introduced after implementing validation.

Comment (krzkaczor) [reply packages/app/src/features/dialogs/savings/deposit/e2e/mainnet/DepositDAI.test-e2e.ts:100]: IMO this is the way to go.

File: packages/app/src/features/dialogs/savings/deposit/e2e/mainnet/DepositDAI.test-e2e.ts [hunk 3]
```diff
@@ -70,3 +71,64 @@ test.describe('Deposit DAI on Mainnet', () => {
...
+  test('actions are disabled', async () => {
+    await depositDialog.fillAmountAction(200)
+    await depositDialog.actionsContainer.expectDisabledActions([
+      { type: 'approve', asset: 'DAI' },
+      { type: 'daiToSDaiDeposit', asset: 'DAI' },
+    ])
+  })
+
+  test('displays sensible tx overview', async () => {
+    await depositDialog.fillAmountAction(10_000)
```
Comment (krzkaczor) [inline packages/app/src/features/dialogs/savings/deposit/e2e/mainnet/DepositDAI.test-e2e.ts:114]: what i don't like about this test is that it doesnt assert that validation failed (it uses a different value).  Also test names are not descriptive enough. I would structurize these tests like this:
* Validation:
  *  When value exceeds balance (here we already input 10_000 in before each):
      * it displays validation error
      * it disables actions
      * it simulates route and apy
  * it displays validation error on zero when dirty

**Review by yivlad — APPROVED** [review]

**Review by krzkaczor — APPROVED** [review]

**Pull request merged**